src/agents/base_agent.py

```
@@ -0,0 +1,161 @@
+            "train_data_buy_and_hold_diff",
+            train_data.iloc[-1][self.main_price_column]
+            - train_data.iloc[0][self.main_price_column],
+        )
+        mlflow.log_param("test_data_shape", test_data.shape)
+        mlflow.log_param(
+            "test_data_buy_and_hold_diff",
+            test_data.iloc[-1][self.main_price_column]
+            - test_data.iloc[0][self.main_price_column],
+        )
+        for i, row in train_data.iterrows():
+            mlflow.log_metric(f"training/data", row[self.main_price_column], step=i)
```

Ruff

f-string without any placeholders

Remove extraneous `f` prefix

(F541)